src/components/ComponentView.vue

```diff
@@ -45,26 +80,57 @@ export default {
         '$route.params.component' (name) {
             this.setupComponent()
         },
+        '$route.query': {
+            handler (name) {
+                this.updateQuery()
+            },
+            deep: true,
+        },
     },
     created () {
         this.setupComponent()
+        this.updateQuery()
     },
     methods: {
         setupComponent () {
             let name = this.$route.params.component
-            let data = components[name]
-            if (!data) {
+            let component = library[name]
+            if (!component) {
                 throw `Component ${name} does not exist!`
             }
 
-            let modelName = data.component.model ? data.component.value : 'value'
-            this.value = data.defaultProps && data.defaultProps[modelName]
+            let modelName = component.model ? component.value : 'value'
+            this.value = component.props[modelName] ? component.props[modelName].default : null
             this.name = name
         },
+        updateQuery () {
+            this.query = { ...this.$route.query }
+        },
         updateValue (value) {
             this.value = value
         },
     },
+    render (h) {
+        if (!this.componentData) {
+            return h()
+        }
+        let slot = this.componentData.slotFn ? this.componentData.slotFn.bind(this.componentData.data)(h) : null
+        if (typeof slot === 'string') {
+            slot = this._v(slot)
+        }
+
+        const propsList = getProps(this.componentData.component).map(prop => {
+            return {
+                ...prop,
+                value: this.componentData.data[prop.name],
+            }
+        })
+
+        // h(SelectProps, { style: { marginRight: '55px' }, props: { propsList } })
```

Comment:
Leftover?

Reply:
Maybe I should just revert ComponentList and ComponentView files since those changes are not needed anymore for snapshot tests to work.